### DISCUSSION TITLE
[Released] ES modules support?

### DISCUSSION THREAD

**Discussion by user72356** [vercel/vercel#8849] [Help] [answered]
I can't import pure esm packages within serverless functions, such as for example "node-fetch" (any pure esm package has the same problem). Please find a minimal example at https://github.com/user72356/vite-project-react

run "npm install", then "vercel dev" and go to http://localhost:3000/api

The console shows the following error:

[ERR_REQUIRE_ESM]: require() of ES Module XXX\vite-project-react\node_modules\node-fetch\src\index.js from XXX\vite-project-react\api\index.ts not supported.

I'm actually using import and not require.

I found this for Next.js apps but I'm not sure that this is being worked for on the Vercel side?

https://nextjs.org/blog/next-11-1#es-modules-support

**Comment by amyegan** [comment]
Hi @user72356! This doesn't look like it's necessarily a Vercel error. It seems to be related to `node-fetch`. There may be a config value that needs to be modified. I found a couple of posts about it on Stack Overflow: https://stackoverflow.com/a/69089164 and https://stackoverflow.com/a/69329950

I get a 404 error when I try to view the repo you linked. But can you share more about the project and what versions of Node and node-fetch you are using? If you can share a working link, I can do some hands-on debugging.

**Reply by user72356** [reply]
Hi @amyegan, sorry about the repo, I did some housekeeping and I think I was overzealous. I used node-fetch as an example because it seems like a widely used ESM package but I have the same problem with any ESM package.

Since my message I have discovered that Vercel was apparently aware of the issue and "sort-of" working on it, but there has not been any progress at all since June. Here is a link to the WIP: https://github.com/vercel/vercel/pull/7954

I know I'm far from the only one with this problem, given the prevalence of Typescript and pure ESM packages these days. I'm hoping you have the power to move this PR forward!!! Thank you very much.

**Reply by amyegan** [reply]
Thanks for sharing that! I'll follow up with the team to find out what's holding up progress on that fix and share updates when I know more.

**Reply by user72356** [reply]
Awesome thank you!

**Reply by amyegan** [reply]
I'm waiting on a specific engineer with historical knowledge around this issue to be back from taking some well earned time off. Once they are back in the office, I'll be able to get more information. Sorry for the delay!

**Comment by amyegan** [comment, marked as answer]
[Support for ESM with TS](https://github.com/vercel/vercel/pull/7954) was released. Note that you may need to make some of the adjustments outlined in https://github.com/vercel/community/discussions/1225#discussioncomment-4642381 to handle some common errors we have seen regarding mixed CJS + ESM use
